sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosResponseDiagnosticsTest extends TestSuiteBase {
+    private CosmosClient gatewayClient;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer cosmosContainer;
+    private CosmosClient directClient;
+
+    private final static String DATABASE_NAME = "TestDiagnosticDB";
+    private final static String CONTAINER_NAME = "TestDiagnosticContainer";
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosResponseDiagnosticsTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() throws Exception {
+        assertThat(this.gatewayClient).isNull();
+        ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.GATEWAY);
+        gatewayClient = clientBuilder().setConnectionPolicy(connectionPolicy).buildClient();
+        createdDatabase = TestSuiteBase.createSyncDatabase(gatewayClient, DATABASE_NAME);
+        CosmosContainerProperties cosmosContainerProperties = new CosmosContainerProperties(CONTAINER_NAME, "/id");
+        cosmosContainer = createdDatabase.createContainer(cosmosContainerProperties).getContainer();
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() throws CosmosClientException {
+        assertThat(this.gatewayClient).isNotNull();
+        createdDatabase.delete();
+        this.gatewayClient.close();
+        if (this.directClient != null) {
+            this.directClient.close();
+        }
+    }
+
+    @Test(groups = {"simple"})
+    public void testGatewayDiagnostics() {
```

Comment:
we don't use "test" as prefix in async sdk. please remove here and for other tests.

Reply:
Done